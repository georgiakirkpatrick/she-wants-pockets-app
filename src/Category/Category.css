.Category {
    width: 46%;
    height: 275px;
}

.Category__cropped {
    /* border-color: black; */
     /* border-width: 2px; */
    /* border-style: solid; */
    /* border-radius: 5px; */
    width: 100%;
    height: 200px;
    object-fit: cover;
    overflow: hidden;
    object-position: 50% 10px ;
}

.Category__cropped img {
    height: 220px;
}

.Category__title {
    color: black;
    /* display: block; */
    font-size: 16px;
    margin-block-start: 0.3em;
    margin-block-end: 0.3em;
    margin-inline-start: 0px;
    margin-inline-end: 0px;
    font-weight: bold;
}

.Category p {
    font-size: 16px;
}

