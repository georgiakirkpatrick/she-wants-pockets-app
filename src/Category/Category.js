import React from 'react'
import { Link } from 'react-router-dom'
import './Category.css'

const Category = props => {
    const makeCategorySlug = () => {
        const categoryName = props.categoryName
        return categoryName
            .normalize("NFD")
            .replace(/[\u0300-\u036f]/g, "")
            .replace(/\s+/g, '-')
            .toLowerCase()
    }

    const slug = makeCategorySlug()

    return (
        <li className='Category' key={props.id}>
            <Link to={`/category/${props.id}/${slug}`} className='main-link'>
                <div className='Category__cropped'>
                    <img src={props.pathToImage} alt={props.imgAlt} />
                </div>
                <h2 className='Category__title'>{props.categoryName}</h2>
            </Link>
        </li>
    )
}

Category.defaultProps = {
    id: 1,
    pathToImage: 'https://media.sezane.com/image/upload/c_fill,d_placeholder_dark.png,fl_progressive:semi,h_816,q_auto:best,w_582/avhc13dzwkv0pzgbmf7g.jpg',
    imgAlt: 'alternate text',
    categoryName: 'Dresses',
}

export default Category