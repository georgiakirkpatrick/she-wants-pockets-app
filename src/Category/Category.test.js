import React from 'react'
import { BrowserRouter as Router } from 'react-router-dom';
import ReactDOM from 'react-dom'
import renderer from 'react-test-renderer'
import Category from './Category'

describe ('Category', () => {
    it ('renders without crashing', () => {
        const div = document.createElement('div')
        ReactDOM.render(
            <Router>
                <Category />
            </Router>
            , div
        )
        ReactDOM.unmountComponentAtNode(div)
    })

    it ('renders the UI as expected', () => {
        const tree = renderer
            .create(<Category />)
            .toJSON()
        expect(tree).toMatchSnapshot()
    })
})