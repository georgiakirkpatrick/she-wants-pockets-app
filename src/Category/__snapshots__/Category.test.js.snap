// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Category renders the UI as expected 1`] = `
<li
  className="Category"
>
  <a
    className="main-link"
    href="/category/1/dresses"
    onClick={[Function]}
  >
    <div
      className="Category__cropped"
    >
      <img
        alt="alternate text"
        src="https://media.sezane.com/image/upload/c_fill,d_placeholder_dark.png,fl_progressive:semi,h_816,q_auto:best,w_582/avhc13dzwkv0pzgbmf7g.jpg"
      />
    </div>
    <h2
      className="Category__title"
    >
      Dresses
    </h2>
  </a>
</li>
`;
